packages/react-native-web/src/exports/Modal/__tests__/index.js

```diff
@@ -0,0 +1,447 @@
+    outsideElement.focus();
+
+    expect(document.activeElement).toBe(outsideElement);
+
+    rerender(
+      <>
+        <a data-testid={'outside'} href={'#outside'}>Outside</a>
+        <Modal visible={true}>
+          <a data-testid={'inside'} href={'#hello'}>Hello</a>
+        </Modal>
+      </>
+    );
+
+    const insideElement = document.querySelector('[data-testid="inside"]');
+    expect(document.activeElement).toBe(insideElement);
+  });
+
+  test('focus is trapped when active', () => {
+    render(
+      <>
+        <a data-testid={'outside'} href={'#outside'}>Outside</a>
+        <Modal visible={true}>
+          <a data-testid={'inside'} href={'#hello'}>Hello</a>
+        </Modal>
+      </>
+    );
+
+    const outsideElement = document.querySelector('[data-testid="outside"]');
+    const insideElement = document.querySelector('[data-testid="inside"]');
+
+    outsideElement.focus();
+
+    expect(document.activeElement).toBe(insideElement);
+  });
+
+  test('focus wraps forwards', () => {
+    render(
+      <>
+        <Modal visible={true}>
+          <a data-testid={'inside-a'} href={'#'}>Inside A</a>
+          <a data-testid={'inside-b'} href={'#'}>Inside B</a>
+          <a data-testid={'inside-c'} href={'#'}>Inside C</a>
+        </Modal>
+      </>
+    );
+
+    const insideStartElement = document.querySelector('[data-testid="inside-a"]');
+    const insideEndElement = document.querySelector('[data-testid="inside-c"]');
+
+    // This is ugly - perhaps there's a better way?
+    const focusBracket = insideEndElement.parentNode.parentNode.parentNode.nextSibling;
+
+    insideEndElement.focus();
+    focusBracket.focus();
+
+    expect(document.activeElement).toBe(insideStartElement);
+  });
+
+  test('focus wraps backwards', () => {
+    render(
+      <>
+        <Modal visible={true}>
+          <a data-testid={'inside-a'} href={'#'}>Inside A</a>
+          <a data-testid={'inside-b'} href={'#'}>Inside B</a>
+          <a data-testid={'inside-c'} href={'#'}>Inside C</a>
+        </Modal>
+      </>
+    );
+
+    const insideStartElement = document.querySelector('[data-testid="inside-a"]');
+    const insideEndElement = document.querySelector('[data-testid="inside-c"]');
+
+    // This is ugly - perhaps there's a better way?
+    const focusBracket = insideEndElement.parentNode.parentNode.parentNode.previousSibling;
+
+    insideStartElement.focus();
+    focusBracket.focus();
+
+    expect(document.activeElement).toBe(insideEndElement);
+  });
+
+  test('focus is trapped without contents', () => {
+    render(
+      <>
+        <a data-testid={'outside'} href={'#outside'}>Outside</a>
+        <Modal visible={true}>
+          <div data-testid={'inside'}>There are no focusable contents.</div>
+        </Modal>
+      </>
+    );
+
+    const outsideElement = document.querySelector('[data-testid="outside"]');;
+
+    outsideElement.focus();
+
+    expect(document.activeElement).toBe(document.body);
```

Comment:
Why is the body the expected active element if the test is expecting focus to be trapped within the modal?

Reply:
Because the modal has nothing within it to focus, so the focus ends up being to the entire document body - preventing an outside element from being focused.

Reply:
I think focus should probably go to the modal. It doesn't really make sense to throw focus to the body and will be confusing for users. The modal can be made programmatically focusable via tabindex -1

Reply:
Sure!

Just to clarify - focus on the `document.body` just mean that nothing is currently focused - we haven't explicitly focused the body.  It's if you take the currently active element and `blur` it. It's really up to the browser how this occurs, but we can override this behavior.

This edge-case [is not particularly WCAG WAI ARIA Modal Best Practices & Guidelines document](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal) so I think that it's a reasonable path to make the modal technically focusable and focus that if there are no other focusable elements within the trap.